tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -10063,7 +10063,29 @@ void hlsl::HandleDeclAttributeForHLSL(Sema &S, Decl *D, const AttributeList &A,
     // The attribute has been set but will have no effect. Validation will emit a diagnostic
     // and prevent code generation.
     ValidateAttributeTargetIsFunction(S, D, A);
+
+    return; // SPIRV Change
+  }
+
+  // SPIRV Change Starts
```

Comment:
Not really.

We need to make sure that even without SPIR-V codegen enabled, HLSL source code with `[[vk::XXX]]` attributes should be accepted by `dxc` as valid (with some warnings maybe). By this way we can guarantee devs use the same source code for both DirectX and Vulkan. So pretty much we need to turn on Vulkan specific modifications in the libraries unconditionally. But for the command line tools, we can conditionally compile features related to SPIR-V.

For this specific case, if we compile conditionally, HLSL source code with `[[vk::location(X)]]` will panic the compiler at https://github.com/Microsoft/DirectXShaderCompiler/blob/ab6759350b1f8550fa522e25154a2aa3d40e05d4/tools/clang/lib/Sema/SemaDeclAttr.cpp#L4610.